File: app/src/main/java/com/github/npcdw/store/util/Sha256Util.java
package com.github.npcdw.store.util;

import java.security.MessageDigest;

public class Sha256Util {
    /**
     * 利用java原生的摘要实现SHA256加密
     */
    public static String getSHA256Str(byte[] bytes) {
        MessageDigest messageDigest;
        String encodeStr = "";
        try {
            messageDigest = MessageDigest.getInstance("SHA-256");
            messageDigest.update(bytes);
            encodeStr = byte2Hex(messageDigest.digest());
        } catch (Exception e) {
            e.printStackTrace();
        }
        return encodeStr;
    }

    /**
     * 将byte转为16进制
     */
    private static String byte2Hex(byte[] bytes) {
        StringBuilder stringBuilder = new StringBuilder();
        String temp = null;
        for (byte aByte : bytes) {
            temp = Integer.toHexString(aByte & 0xFF);
            if (temp.length() == 1) {
                // 1得到一位的进行补0操作
                stringBuilder.append("0");
            }
            stringBuilder.append(temp);
        }
        return stringBuilder.toString();
    }
}

File: app/src/main/java/com/github/npcdw/store/util/Sha256Util.kt
package com.github.npcdw.store.util

import java.security.MessageDigest

object Sha256Util {
    /**
     * 利用java原生的摘要实现SHA256加密
     */
    fun getSHA256Str(bytes: ByteArray?): String {
        val messageDigest: MessageDigest
        var encodeStr = ""
        try {
            messageDigest = MessageDigest.getInstance("SHA-256")
            messageDigest.update(bytes)
            encodeStr = byte2Hex(messageDigest.digest())
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return encodeStr
    }

    /**
     * 将byte转为16进制
     */
    private fun byte2Hex(bytes: ByteArray): String {
        val stringBuilder = StringBuilder()
        var temp: String?
        for (aByte in bytes) {
            temp = Integer.toHexString(aByte.toInt() and 0xFF)
            if (temp.length == 1) {
                // 1得到一位的进行补0操作
                stringBuilder.append("0")
            }
            stringBuilder.append(temp)
        }
        return stringBuilder.toString()
    }
}
